frontend-wasm/src/miden_abi/transform.rs

```diff
@@ -0,0 +1,101 @@
+use miden_diagnostics::DiagnosticsHandler;
+use miden_hir::{FunctionIdent, InstBuilder, SourceSpan, Type::*, Value};
+
+use super::tx_kernel;
+use crate::module::function_builder_ext::FunctionBuilderExt;
+
+/// The strategy to use for transforming a function call
+enum TransformStrategy {
+    /// The Miden ABI function returns a length and a pointer and we only want the length
+    ListReturn,
+    /// The Miden ABI function returns on the stack and we want to return via a pointer argument
+    ReturnViaPointer,
+    /// No transformation needed
+    NoTransform,
+}
+
+/// Get the transformation strategy for a function name
+fn get_transform_strategy(function_id: &str) -> TransformStrategy {
+    match function_id {
+        tx_kernel::NOTE_GET_INPUTS => TransformStrategy::ListReturn,
+        tx_kernel::ACCOUNT_ADD_ASSET => TransformStrategy::ReturnViaPointer,
+        tx_kernel::ACCOUNT_GET_ID => TransformStrategy::NoTransform,
+        _ => panic!("No transform strategy found for function {}", function_id),
+    }
+}
+
+/// Transform a function call based on the transformation strategy
+pub fn transform_miden_abi_call(
+    func_id: FunctionIdent,
+    args: &[Value],
+    builder: &mut FunctionBuilderExt,
+    span: SourceSpan,
+    diagnostics: &DiagnosticsHandler,
+) -> Vec<Value> {
+    use TransformStrategy::*;
+    match get_transform_strategy(func_id.function.as_symbol().as_str()) {
+        ListReturn => list_return(func_id, args, builder, span, diagnostics),
+        ReturnViaPointer => return_via_pointer(func_id, args, builder, span, diagnostics),
+        NoTransform => no_transform(func_id, args, builder, span, diagnostics),
+    }
+}
+
+/// No transformation needed
+#[inline(always)]
+pub fn no_transform(
+    func_id: FunctionIdent,
+    args: &[Value],
+    builder: &mut FunctionBuilderExt,
+    span: SourceSpan,
+    _diagnostics: &DiagnosticsHandler,
+) -> Vec<Value> {
+    let call = builder.ins().call(func_id, args, span);
+    let results = builder.inst_results(call);
+    results.to_vec()
+}
+
+/// The Miden ABI function returns a length and a pointer and we only want the length
+pub fn list_return(
```

Comment:
I think this is basically a variant of `no_transform` right? Discarding an unused return value is totally fine with regular functions.

Again, if we're not emitting a wrapper function or something in these cases, feel free to ignore, just flagging it in case the assumption was that we needed to do something different for these functions.

I will definitely make a note though to double check how we're handling multi-result functions elsewhere in the compiler, just in case this is necessary due to lack of support later in the pipeline.

Reply:
I'm not sure, I'm following. `list_return` discards the second return value, i.e. transforming the result to adopt it to the expected Wasm stack state where only one return value is expected. So in my mental model it's not `no_transform`. 
To clarify, there are no wrapper functions generated during the transformation process. The transformation code is generated at the call site.

Reply:
Ok, you can disregard this then, my note was based on the premise that we were emitting wrapper functions to adapt the ABI differences, but if we're just generating code at each call site, then you're already doing what I was suggesting